collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -210,7 +178,7 @@ public void onCreate(Bundle savedInstanceState) {
 
             MyDrive = false;
 
-            if (testNetwork()) {
+            if (isDeviceOnline()) {
```

Comment:
I am pretty sure it should be `isDeviceOnline()`. 
```
/**
     * Checks whether the device currently has a network connection.
     *
     * @return true if the device has a network connection, false otherwise.
     */
    private boolean isDeviceOnline() {
        ConnectivityManager connMgr =
                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
        return (networkInfo != null && networkInfo.isConnected());
    }
```
I have tested on device too and it's working as it should. Can you please recheck if its not working or not on your device a well?